.github/workflows/deploy.yml

```diff
@@ -0,0 +1,33 @@
+name: Shopify Analytics Deploy
+
+on:
+  push:
+    branches: [ master ]
+  # Allow running this workflow manually from the Actions tab
+  workflow_dispatch:
+
+env:
+  ECR_IMAGE_REPO: "884681002735.dkr.ecr.us-east-1.amazonaws.com/shopify-analytics"
+  DEPLOY_IAM_ROLE: "arn:aws:iam::884681002735:role/github_actions/github-actions-role"
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Build Image
+        run: docker build -t ${{ env.ECR_IMAGE_REPO }}:${{ github.sha }} --label org.label-schema.name="shopify-analytics" .
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ env.DEPLOY_IAM_ROLE }}
+          aws-region: us-east-1
```

Comment:
I think you need to specify special permissions to log into aws. Something like:

```
      permissions:
        id-token: write
        contents: read
```

Ref: https://github.com/aws-actions/configure-aws-credentials#oidc

Reply:
Good catch! Thats bit me before too:

Thanks!

Reply:
https://github.com/iFixit/shopify-analytics/pull/13/commits/8aeb3f67dfd0662af73954993cddec5342c07cf1